api_tests/collections/test_views.py

```diff
@@ -3906,7 +3906,7 @@ def test_linked_preprints_returns_everything(
         res = app.get(url_collection_linked_preprints, auth=user_one.auth)
 
         assert res.status_code == 200
-        preprints_returned = [linked_preprint['id']
+        preprints_returned = [linked_preprint['id'].split(VersionedGuidMixin.GUID_VERSION_DELIMITER)[0]
```

Comment:
I think the test failure may review an actual bug in collection (or in the test fixture). We shouldn't chop off the version number to make the test pass. Here is an example:

```
____ TestCollectionLinkedPreprints.test_linked_preprints_returns_everything ____
api_tests/collections/test_views.py:3914: in test_linked_preprints_returns_everything
    assert preprint_returned in id_linked_preprints
E   AssertionError: assert 'b7nvp_v1' in ['b7nvp', 'tmzkx', 'zm6up']

```

For the above failure, I think the correct fix is to fix `id_linked_preprints` so that `assert 'b7nvp_v1' in ['b7nvp_v1', 'tmzkx_v1' pass; rather than 'zm6up_v1']` rather than chop of the version in`preprint_returned` and pass with `assert 'b7nvp' in ['b7nvp_v1', 'tmzkx_v1', 'zm6up_v1']`.

Maybe it's an issue with the fixture `id_linked_preprints`; maybe it's an issue with the `.guids_link`; and/or it could even be an issue with the `.collect_object()`. Please take a look.